core/types/transaction.go

```diff
@@ -337,7 +337,7 @@ func (tx *Transaction) WithSignature(signer Signer, sig []byte) (*Transaction, e
 		return nil, err
 	}
 	cpy := tx.inner.copy()
-	cpy.setSignatureValues(v, r, s)
+	cpy.setSignatureValues(signer.ChainID(), v, r, s)
```

Comment:
Shouldn't this method also raise an error if `signer.ChainID` differs from `tx.ChainID` ?

Reply:
The check is done in SignatureValues, so the error will be raised from there.